media-proxy/include/mesh/st2110rx.h

```diff
@@ -0,0 +1,174 @@
+                         const MeshConfig_ST2110& cfg_st2110) {
+        int session_id = 0;
+        mtl_device = get_mtl_device(dev_port, MTL_LOG_LEVEL_CRIT, cfg_st2110.local_ip_addr, session_id);
+        if (mtl_device == nullptr) {
+            log::error("Failed to get MTL device");
+            return -1;
+        }
+
+        inet_pton(AF_INET, cfg_st2110.remote_ip_addr, ops.port.ip_addr[MTL_PORT_P]);
+        inet_pton(AF_INET, cfg_st2110.local_ip_addr, ops.port.mcast_sip_addr[MTL_PORT_P]);
+        ops.port.udp_port[MTL_PORT_P] = cfg_st2110.local_port;
+
+        strlcpy(ops.port.port[MTL_PORT_P], dev_port.c_str(), MTL_PORT_MAX_LEN);
+        ops.port.num_port = 1;
+
+        char session_name[NAME_MAX] = "";
+        snprintf(session_name, NAME_MAX, "mcm_mtl_rx_%d", session_id);
+        if (ops.name)
+            free((void *)ops.name);
+        ops.name = strdup(session_name);
+        ops.framebuff_cnt = 4;
+
+        ops.priv = this; // app handle register to lib
+        ops.notify_frame_available = frame_available_cb;
+
+        log::info("ST2110Rx: configure")("port", ops.port.port[MTL_PORT_P])(
+            "ip_addr", std::to_string(ops.port.ip_addr[MTL_PORT_P][0]) + " " +
+                           std::to_string(ops.port.ip_addr[MTL_PORT_P][1]) + " " +
+                           std::to_string(ops.port.ip_addr[MTL_PORT_P][2]) + " " +
+                           std::to_string(ops.port.ip_addr[MTL_PORT_P][3]))(
+            "mcast_sip_addr", std::to_string(ops.port.mcast_sip_addr[MTL_PORT_P][0]) + " " +
+                                  std::to_string(ops.port.mcast_sip_addr[MTL_PORT_P][1]) + " " +
+                                  std::to_string(ops.port.mcast_sip_addr[MTL_PORT_P][2]) + " " +
+                                  std::to_string(ops.port.mcast_sip_addr[MTL_PORT_P][3]))(
+            "num_port", ops.port.num_port)("udp_port", ops.port.udp_port[MTL_PORT_P])(
+            "name", ops.name)("framebuff_cnt", ops.framebuff_cnt);
+
+        return 0;
+    }
+
+    Result on_establish(context::Context& ctx) override {
+        _ctx = context::WithCancel(ctx);
+        stop = false;
+
+        mtl_session = create_session(mtl_device, &ops);
+        if (!mtl_session) {
+            log::error("Failed to create session");
+            set_state(ctx, State::closed);
+            return set_result(Result::error_general_failure);
+        }
+
+        /* Start MTL session thread. */
+        try {
+            frame_thread_handle = std::jthread(&ST2110Rx::frame_thread, this);
+        } catch (const std::system_error& e) {
+            log::error("Failed to create thread");
+            set_state(ctx, State::closed);
+            return set_result(Result::error_out_of_memory);
+        }
+
+        set_state(ctx, State::active);
+        return set_result(Result::success);
+    }
+
+    Result on_shutdown(context::Context& ctx) override {
+        _ctx.cancel();
+        stop = true;
+        stop.notify_all();
```

Comment:
```suggestion
        frame_available.store(true, std::memory_order_release);
        frame_available.notify_one();
```

Reply:
Fixed in 133a6efcdea8895252fd1a825a2ae57a7f31cb01